src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,104 @@
+package mate.academy.dao;
+
+import mate.academy.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.util.ConnectionUtil;
+import mate.academy.model.Book;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements  BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection
+                     .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Need insert at lest one row");
+            }
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if(resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                book.setId(id);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ? AND is_deleted = FALSE";
+        Book book = new Book();
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = getBookFormResult(resultSet);
+            }
+            return Optional.ofNullable(book);
```

Comment:
can book be null in your code?
![image](https://github.com/mate-academy/jv-jdbc-intro/assets/20326858/8b1e9a1a-2b29-4715-afa5-c4d2a27454e9)